File: rabbitmq-poc/src/main/java/com/example/demo/RabbitMQConfig.java
package com.example.demo;

import org.springframework.amqp.core.Binding;
import org.springframework.amqp.core.BindingBuilder;
import org.springframework.amqp.core.Exchange;
import org.springframework.amqp.core.ExchangeBuilder;
import org.springframework.amqp.core.Queue;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.beans.factory.annotation.Value;

@Configuration
public class RabbitMQConfig {

    @Value("${app.queue.name}")
    private String queueName;

    @Value("${app.exchange.name}")
    private String exchangeName;

    @Value("${app.routing-key}")
    private String routingKey;

    @Bean
    public Queue appQueue() {
        return new Queue(queueName, true);
    }

    @Bean
    public Exchange appExchange() {
        return ExchangeBuilder.topicExchange(exchangeName).durable(true).build();
    }

    @Bean
    public Binding appBinding(Queue appQueue, Exchange appExchange) {
        return BindingBuilder.bind(appQueue).to(appExchange).with(routingKey).noargs();
    }
}

File: rabbitmq-poc/src/main/java/com/example/demo/RabbitMQConfig.kt
package com.example.demo

import org.springframework.amqp.core.*
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class RabbitMQConfig(
    @Value("\${app.queue.name}")
    private val queueName: String,
    @Value("\${app.exchange.name}")
    private val exchangeName: String,
    @Value("\${app.routing-key}")
    private val routingKey: String
) {

    @Bean
    fun appQueue(): Queue = Queue(queueName, true)

    @Bean
    fun appExchange(): Exchange = ExchangeBuilder.topicExchange(exchangeName).durable(true).build()

    @Bean
    fun appBinding(appQueue: Queue, appExchange: Exchange): Binding =
        BindingBuilder.bind(appQueue).to(appExchange).with(routingKey).noargs()
}
